File: client-scalaz/src/main/scala/org/openapitools/client/api/RetailerLocationResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import RetailerLocationResponse._

case class RetailerLocationResponse (
  retailerLocationId: Option[Long],
name: Option[String],
contact: Option[ContactInfoResponse],
latitude: Option[Double],
longitude: Option[Double],
active: Option[Boolean],
favoriteId: Option[Long],
favorite: Option[Boolean],
responsible: Option[ProfileShortResponse],
manager: Option[ProfileShortResponse],
categories: Option[List[CategoryResponse]],
filters: Option[List[FilterResponse]],
logo: Option[AssetShortResponse],
picture1: Option[AssetShortResponse],
picture2: Option[AssetShortResponse],
billableEntity: Option[BillableEntityResponse],
counts: Option[RetailerCountResponse],
retailer: Option[RetailerShortResponse],
offers: Option[ListResponse],
internalId: Option[String],
detailsHeader: Option[String],
detailsBody: Option[String],
hours: Option[String],
locationToken: Option[String],
building: Option[String],
qrCodeUrl: Option[String],
distance: Option[Double],
googlePlaceId: Option[String],
yelpId: Option[String])

object RetailerLocationResponse {
  import DateTimeCodecs._

  implicit val RetailerLocationResponseCodecJson: CodecJson[RetailerLocationResponse] = CodecJson.derive[RetailerLocationResponse]
  implicit val RetailerLocationResponseDecoder: EntityDecoder[RetailerLocationResponse] = jsonOf[RetailerLocationResponse]
  implicit val RetailerLocationResponseEncoder: EntityEncoder[RetailerLocationResponse] = jsonEncoderOf[RetailerLocationResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/retailer_location_response.clj
(ns sirqul-io-t-platform.specs.retailer-location-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.contact-info-response :refer :all]
            [sirqul-io-t-platform.specs.profile-short-response :refer :all]
            [sirqul-io-t-platform.specs.profile-short-response :refer :all]
            [sirqul-io-t-platform.specs.category-response :refer :all]
            [sirqul-io-t-platform.specs.filter-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            [sirqul-io-t-platform.specs.billable-entity-response :refer :all]
            [sirqul-io-t-platform.specs.retailer-count-response :refer :all]
            [sirqul-io-t-platform.specs.retailer-short-response :refer :all]
            [sirqul-io-t-platform.specs.list-response :refer :all]
            )
  (:import (java.io File)))


(def retailer-location-response-data
  {
   (ds/opt :retailerLocationId) int?
   (ds/opt :name) string?
   (ds/opt :contact) contact-info-response-spec
   (ds/opt :latitude) float?
   (ds/opt :longitude) float?
   (ds/opt :active) boolean?
   (ds/opt :favoriteId) int?
   (ds/opt :favorite) boolean?
   (ds/opt :responsible) profile-short-response-spec
   (ds/opt :manager) profile-short-response-spec
   (ds/opt :categories) (s/coll-of category-response-spec)
   (ds/opt :filters) (s/coll-of filter-response-spec)
   (ds/opt :logo) asset-short-response-spec
   (ds/opt :picture1) asset-short-response-spec
   (ds/opt :picture2) asset-short-response-spec
   (ds/opt :billableEntity) billable-entity-response-spec
   (ds/opt :counts) retailer-count-response-spec
   (ds/opt :retailer) retailer-short-response-spec
   (ds/opt :offers) list-response-spec
   (ds/opt :internalId) string?
   (ds/opt :detailsHeader) string?
   (ds/opt :detailsBody) string?
   (ds/opt :hours) string?
   (ds/opt :locationToken) string?
   (ds/opt :building) string?
   (ds/opt :qrCodeUrl) string?
   (ds/opt :distance) float?
   (ds/opt :googlePlaceId) string?
   (ds/opt :yelpId) string?
   })

(def retailer-location-response-spec
  (ds/spec
    {:name ::retailer-location-response
     :spec retailer-location-response-data}))
